# HG changeset patch
# User jwe
# Date 745017448 0
#      Tue Aug 10 21:17:28 1993 +0000
# Node ID 969b6494a10ca3c8c86f7fe42c64961be35975c1
# Parent  c63068b25a6318c7446384ed675c3a2083ec92a3
[project @ 1993-08-10 21:17:28 by jwe]
Use AC_HAVE_LIBRARY to check for -lm.

diff --git a/configure.in b/configure.in
--- a/configure.in
+++ b/configure.in
@@ -297,21 +297,19 @@ case "$target_host_type" in
 esac
 
 # Check to see if we have IEEE math functions, and if so, which ones.
 #
 # Also check for some additional trig functions that aren't ANSI but
 # are often available.  If they are missing, we try to replace them
 # with functions from the BSD/NET2 math library.
 
-XLIBS="$LIBS"
-LIBS="$XLIBS -lm"
+AC_HAVE_LIBRARY(-lm)dnl
 AC_HAVE_FUNCS(finite isnan isinf)dnl
 AC_REPLACE_FUNCS(acosh asinh atanh)dnl
-LIBS="$XLIBS"
 
 if test -n "$LIBOBJS"; then
   for func in $LIBOBJS
   do
     case "$func" in
       *acosh.o*)
 	DEFS="$DEFS -DACOSH_MISSING=1"
 	;;
@@ -366,30 +364,22 @@ warning: I didn't find gnuplot.  It isn'
          tell Octave where to find it by typing the command
 
            gnuplot_binary = '/full/path/to/gnuplot/binary'
 
          at the Octave prompt.
 EOF
 fi
 
-AC_PROGRAM_CHECK(DEFAULT_PAGER, less, less, [])dnl
-if test -z "$DEFAULT_PAGER"; then
-  AC_PROGRAM_CHECK(DEFAULT_PAGER, more, more, [])dnl
-fi
-if test -z "$DEFAULT_PAGER"; then
-  AC_PROGRAM_CHECK(DEFAULT_PAGER, page, page, [])dnl
-fi
-if test -z "$DEFAULT_PAGER"; then
-  AC_PROGRAM_CHECK(DEFAULT_PAGER, pg, pg, [])dnl
-fi
+AC_PROGRAMS_CHECK(DEFAULT_PAGER, less more page pg, [])dnl
 if test -z "$DEFAULT_PAGER"; then
   echo "warning: I couldn't find less(1), more(1), page(1), or pg(1)"
+elif test "$DEFAULT_PAGER" = less; then
+  DEFAULT_PAGER="less -e"
 fi
-AC_SUBST(DEFAULT_PAGER)dnl
 
 AC_PROGRAM_CHECK(RUNTEST, runtest, runtest, [])dnl
 AC_PROGRAM_CHECK(EXPECT, expect, expect, [])dnl
 if test -z "$EXPECT" ; then
   RUNTEST=
 fi
 AC_SUBST(RUNTEST)dnl
 
